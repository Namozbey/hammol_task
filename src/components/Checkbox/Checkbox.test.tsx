import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Checkbox } from ".";

describe("Testing Checkbox", () => {
  it("Snapshots", () => {
    const { asFragment } = render(<Checkbox name="checkbox" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("Fire change event", () => {
    const { getByTestId } = render(
      <Checkbox id="checkbox" data-testid="checkbox" />
    );
    const element = getByTestId("checkbox") as HTMLInputElement;
    expect(element.checked).toBe(false);

    fireEvent.change(element, { target: { checked: true } });
    expect(element.checked).toBe(true);
  });
});
