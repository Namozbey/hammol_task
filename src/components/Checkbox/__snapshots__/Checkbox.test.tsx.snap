// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Checkbox Snapshots 1`] = `
<DocumentFragment>
  <input
    class="sc-bczRLJ iWjHhS"
    name="checkbox"
    type="checkbox"
  />
</DocumentFragment>
`;
